Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -158,7 +158,15 @@ extension NavigationMapView {
         /**
          Take the remaining distance from the upcoming point on the route and extends it by the exact position of the puck.
          */
-        let remainingDistance = traveledIndex.distanceRemaining + calculateDistance(coordinate1: upcomingPoint, coordinate2: location)
```

Comment:
This one I want more feedback on. 

**The problem: ** When the progress proceed after the final point, `traveledIndex.distanceRemaining` become zero, but `calculateDistance(coordinate1: upcomingPoint, coordinate2: location)` starts to grow for obvious reasons: distance is always positive.    

**Solution: ** Check if the progress is at final leg, user arrived at the waypoint and the distance remaining is 0. First two checks are straightforward, but the second one is debatable. I've observed that when currentLegProgress.distanceRemaining reaches 0 it doesn't change anymore. And that's what we want as far as I see. If anybody else knows a better check, I'm more than happy to change this code.